js/ui/status/automaticUpdates.js

```diff
@@ -0,0 +1,301 @@
+
+        });
+        this._item.menu.addSettingsAction(_("Set a Schedule…"), 'gnome-updates-panel.desktop');
+        this.menu.addMenuItem(this._item);
+
+        this._activeConnection = null;
+
+        NM.Client.new_async(null, Lang.bind(this, this._clientGot));
+    },
+
+    _clientGot: function(obj, result) {
+        this._client = NM.Client.new_finish(result);
+
+        this._client.connect('notify::activating-connection', Lang.bind(this, this._sync));
+        this._client.connect('notify::primary-connection', Lang.bind(this, this._sync));
+
+        this._sync();
+
+        Main.sessionMode.connect('updated', Lang.bind(this, this._sessionUpdated));
+        this._sessionUpdated();
+
+        // Start retrieving the Mogwai proxy
+        this._proxy = new SchedulerProxy(Gio.DBus.system,
+                                         'com.endlessm.MogwaiScheduler1',
+                                         '/com/endlessm/DownloadManager1',
+                                          Lang.bind(this, function(proxy, error) {
+                                              if (error) {
+                                                  log(error.message);
+                                                  return;
+                                              }
+                                              this._proxy.connect('g-properties-changed',
+                                                                  Lang.bind(this, this._sync));
+                                              this._sync();
+                                          }));
+    },
+
+    _sessionUpdated: function() {
+        let sensitive = !Main.sessionMode.isLocked && !Main.sessionMode.isGreeter;
+        this.menu.setSensitive(sensitive);
+    },
+
+    _sync: function() {
+
+        if (!this._client || !this._proxy)
+            return;
+
+        // Update the current active connection. This will connect to the
+        // NM.SettingUser signal to sync every time someone updates the
+        // NM_SETTING_ALLOW_DOWNLOADS_WHEN_METERED setting.
+        this._updateActiveConnection();
+
+        // Toggle item name
+        this._updateAutomaticUpdatesItem();
+
+        // Icons
+        let icon = this._getIcon()
+
+        this._item.icon.gicon = icon;
+        this._indicator.gicon = icon;
+
+        // Only show the Automatic Updates icon at the bottom bar when it is
+        // both enabled, and there are updates being downloaded or installed.
+        this._updateVisibility();
+
+        // The status label
+        this._item.label.text = _("Automatic Updates");
+    },
+
+    _updateActiveConnection: function() {
+        let currentActiveConnection = this._getActiveConnection();
+
+        if (this._activeConnection == currentActiveConnection)
+            return;
+
+        // Disconnect from the previous active connection
+        if (this._settingChangedSignalId)
+            this._activeConnection.disconnect(this._userSettingSignalId);
```

Comment:
This should be `_settingChangedSignalId`, no?

Reply:
Nice catch.